### ISSUE TITLE
Add Docker support

### ISSUE THREAD

**Issue by marcelaraujo** [firehol/netdata#84]
Please add Dockerfile to be able create containers.

Tks

**Comment by Veve2** [comment]
Here is the repository on dockerhub : https://hub.docker.com/r/titpetric/netdata/

And here is the coresponding dockerfile :+1: 

```
FROM debian:jessie RUN DEBIAN_FRONTEND=noninteractive apt-get -qq update
RUN DEBIAN_FRONTEND=noninteractive apt-get -qq -y install zlib1g-dev gcc make git autoconf autogen automake pkg-config

ADD build.sh /build.sh
ADD run.sh /run.sh
RUN chmod +x /run.sh /build.sh; sync; sleep 1; /build.sh

WORKDIR /

ENTRYPOINT ["/run.sh"]
```

**Comment by marcelaraujo** [comment]
Thank you.

**Comment by titpetric** [comment]
Source is (currently) available on my github repo list. Soon it will be merged to the main repo, that's the plan.

**Comment by ktsaou** [comment]
duplicate of #10 with PR #77

**Comment by ROODAY** [comment]
@titpetric Is there a way to mount a config file to the docker container? And is the docker netdata container aware of the resources being used by docker and my other docker containers?

**Comment by titpetric** [comment]
@ROODAY

Hello,
1) it should be (docker -v option)
2) technically, anything under proc is readable, but not very... Human readable. You will not get container names, etc.
3) docker pull. Generally the image is updated at least weekly.

Unless you only care about system/network metrics, running netdata in docker is more of a "try before you commit to it" type of deal. Exposing user land services (apache, nginx, mysql...) to a docker container is... Extremely difficult. Installation from a package or git is recommended.

**Comment by ROODAY** [comment]
@titpetric 

Ah. The problem is I have a Digital Ocean server with all my applications running in docker containers, including an nginx reverse proxy, and it'd be hassle to have netdata outside of that setup.

**Comment by titpetric** [comment]
@ROODAY I have the same. Realize that the biggest issue that containers are isolated, including netdata. You can have a container that has connectivity to other containers ([How to live inside a docker container](https://scene-si.org/2016/04/01/how-to-live-inside-a-docker-container/)), but that doesn't expose the needed resources from those containers to netdata. They live in isolation.

Metrics which are available on the host (/proc filesystem) are the most reliable to deliver to a running netdocker container. Surface metrics for docker containers will be available with time (work is being done in #115), application metrics are already available via the apps plugin (you just don't know which container an individual app belongs to).

**Comment by ROODAY** [comment]
@titpetric Ok, thanks for the info!
